src/main/java/com/mongodb/jdbc/MongoDriver.java

```diff
@@ -172,7 +172,9 @@ public Connection connect(String url, Properties info) throws SQLException {
             try {
                 conn.testConnection(conn.getDefaultConnectionValidationTimeoutSeconds());
             } catch (TimeoutException te) {
-                throw new SQLTimeoutException("Timeout. Can't connect.");
+                throw new SQLTimeoutException(
+                        "Couldn't connect due to a timeout. Please check your host and port, and if necessary set a "
+                                + "longer connection timeout in the MongoDB URI.");
```

Comment:
```suggestion
                        "Couldn't connect due to a timeout. Please check your hostname and port. If necessary set a "
                                + "longer connection timeout in the MongoDB URI.");
```
Suggestion for the error message.